drivers/tee/optee/rpmb.c

```diff
@@ -131,8 +142,23 @@ static u32 rpmb_process_request(struct optee_private *priv, void *req,
 		mmc = get_mmc(priv, sreq->dev_id);
 		if (!mmc)
 			return TEE_ERROR_ITEM_NOT_FOUND;
-		if (mmc_rpmb_route_frames(mmc, RPMB_REQ_DATA(req),
-					  req_size - sizeof(struct rpmb_req),
+
+		rpmb_frame_data_sz = (req_size - sizeof(struct rpmb_req));
+		if ((!rpmb_frame_buffer) || (rpmb_frame_data_sz > rpmb_frame_buffer_sz))
+		{
+			if (rpmb_frame_buffer)
+				free(rpmb_frame_buffer);
+
+			if (rpmb_frame_data_sz > rpmb_frame_buffer_sz)
+				rpmb_frame_buffer_sz = rpmb_frame_data_sz;
+
+			rpmb_frame_buffer = malloc(rpmb_frame_buffer_sz);
+			if (!rpmb_frame_buffer)
+				return TEE_ERROR_OUT_OF_MEMORY;
+		}
+		memcpy(rpmb_frame_buffer, RPMB_REQ_DATA(req), rpmb_frame_data_sz);
```

Comment:
this allocates and copy bffer even if rpmb frame buffer is 32bit aligned. In such case, we could pass it straight to `mmc_rpmb_route_frames()`.

Reply:
I fixed this, but it adds a big block in a previously seamless code, would it be worth it to create another function to get the aligned address ?
Which would return `RPMB_REQ_DATA(req)` if the alignment is correct, and if not creates the buffer if needed and copy the data in it.
It would return a void* to the address of either the buffer or the data after the request
Something like `void* get_aligned_rpmb_data(void* addr, size_t data_len)`